.github/workflows/main.yml

```diff
@@ -0,0 +1,34 @@
+name: CI/CD Pipeline
+
+on:
+  push:
+    branches:
+      - '**'
```

Comment:
We don’t want our GitHub workflows to run every time someone pushes code. Instead, we only want them to run when a pull request is created or updated.  

Please remove the `push` trigger from the workflow configuration and keep only the `pull_request` trigger. This way, workflows will only run when someone opens a pull request or adds new commits to an existing one, which helps avoid unnecessary runs and keeps the workflow usage more efficient.